### PR TITLE
JETUT-72: rewrote web profile introduction

### PR DESCRIPTION
**Pull request by BalusC**
https://virtua.atlassian.net/browse/JETUT-72

I opted to not rename/move/delete adoc files yet. Will be done once ordering is set in stone.

### CODE REVIEW

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
-The configuration information can be specified using Jakarta EE annotations or can be maintained in a text file in XML format called a web application deployment descriptor (DD).
-A web application DD must conform to the schema described in the Jakarta Servlet specification.
-
-This chapter gives a brief overview of the activities involved in developing web applications.
-First, it summarizes the web application lifecycle and explains how to package and deploy very simple web applications on GlassFish Server.
-The chapter then moves on to configuring web applications and discusses how to specify the most commonly used configuration parameters.
+. The client sends an HTTP request to the web server.
+. The web server module that supports Jakarta Servlet based web components is called a servlet container.
+. The servlet container converts the HTTP request into an `HttpServletRequest` object and prepares the `HttpServletResponse` object.
+. These objects are delivered to a web component, which can interact with JavaBeans components or a database to generate dynamic content.
```
Comment (BalusC) [inline src/main/asciidoc/webapp/webapp001.adoc:29]: Personally not a fan of wording "JavaBeans components". How is this term used in CDI chapters?

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 2]
```diff
@@ -1,32 +1,130 @@
...
+ |              |    `-- js
+ |              |         `-- script.js
+ |              |-- WEB-INF
+ |              |    |-- beans.xml
+ |              |    |-- faces-config.xml
+ |              |    `-- web.xml
+ |              |-- favicon.ico
+ |              `-- index.xhtml
+ `-- pom.xml
+----
```
Comment (BalusC) [inline src/main/asciidoc/webapp/webapp002.adoc:70]: Probably a screenshot of a folder structure in Eclipse IDE is a better idea?

**Comment by arjantijms** [comment]
Looks good, the term "web component" as part of the archaic "application client component, applet component, web component, ejb component" is maybe debatable, but the spec still defines it and it's used all over the place.

**Review by arjantijms — APPROVED** [review]

**Review by kito99 — CHANGES_REQUESTED** [review]

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 2 above)
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:77]: Okay :)

**Comment by BalusC** [comment]
> Looks good, the term "web component" as part of the archaic "application client component, applet component, web component, ejb component" is maybe debatable, but the spec still defines it and it's used all over the place.

I also had my doubts on this term but it's indeed used all over the place including images.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 3]
```diff
@@ -1,46 +1,52 @@
...
-This object is delivered to a web component, which can interact with JavaBeans components or a database to generate dynamic content.
-The web component can then generate an `HTTPServletResponse` or can pass the request to another web component.
-A web component eventually generates a `HTTPServletResponse` object.
-The web server converts this object to an HTTP response and returns it to the client.
+A service-oriented web application generates dynamic data structures in response to HTTP requests.
+The response is usually represented as a JSON object or as a XML document.
+Service-oriented applications are often directly used by presentation-oriented web applications.
+Development of service-oriented web applications is covered in [TBD].
+
+In the Jakarta EE platform, web applications are represented by web components as seen in xref:jakarta-web-application-request-handling[xrefstyle=short].
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:20]: I've renamed "JavaBeans components" to "Beans" in the image.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 4]
```diff
@@ -1,46 +1,52 @@
...
+A web application, also known as a web app, is a software application that runs on one or more web servers.
+It is typically accessed through a web browser over a network, such as the Internet.
+The advantage over desktop and mobile applications is being platform-independent, as it can be accessed and used on different devices.
 Web applications are of the following types:
 
 Presentation-oriented::
-A presentation-oriented web application generates interactive web pages containing various types of markup language (HTML, XHTML, XML, and so on) and dynamic content in response to requests.
-Development of presentation-oriented web applications is covered in xref:jakarta-faces-technology-2[xrefstyle=full] through xref:jakarta-servlet-technology-2[xrefstyle=full]
+A presentation-oriented web application generates dynamic web pages in response to HTTP requests.
+The response is usually represented as a HTML document along with assets.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:10]: The term "assets" is expanded in next chapter. But I think you're right to expand it sooner.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 5]
```diff
@@ -1,46 +1,52 @@
...
+Presentation-oriented applications are often directly used by humans.
+Development of presentation-oriented web applications is covered in [TBD].
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:12]: Ah okay!

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 6]
```diff
@@ -1,46 +1,52 @@
...
 
 Service-oriented::
-A service-oriented web application implements the endpoint of a web service.
-Presentation-oriented applications are often clients of service-oriented web applications.
-Development of service-oriented web applications is covered in xref:building-web-services-with-jakarta-xml-web-services[xrefstyle=full] and xref:building-restful-web-services-with-jakarta-rest[xrefstyle=full] in xref:web-services[xrefstyle=full]
-
-In the Jakarta EE platform, web components provide the dynamic extension capabilities for a web server.
-Web components can be Jakarta servlets, web pages implemented with Jakarta Faces technology, web service endpoints, or Jakarta Server Pages.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:16]: Hmm you're commenting on a block that's deleted/modified?

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 7]
```diff
@@ -1,32 +1,140 @@
...
+The process for creating, deploying, and executing a Jakarta Servlet based web application is different from that of Java classes which are packaged and executed as a Java application archive (JAR).
+It can be summarized as follows:
 
 . Develop the web component code.
+. Develop the deployment descriptor files, if necessary.
+. Compile the web component code against the libraries of the servlet container and the helper libraries, if any.
+. Package the compiled code along with helper libraries, assets and deployment descriptor files, if any, into a deployable unit, called a web application archive (WAR).
+. Deploy the WAR into a servlet container.
+. Run the web application by accessing a URL that references the web component.
+
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:19]: This is the 'Web Profile' section. REST in stand-alone JAR is supposed to be detailed in 'Core Profile' section.

File: src/main/asciidoc/webapp/webapp001.adoc
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:1]: Ok, I've added a hyphen before 'based' all over place.

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 8]
```diff
@@ -1,32 +1,140 @@
...
+You need Maven and a servlet container.
+
+==== Building The Example Projects
+
+This describes steps 3 and 4.
+Maven is used to build the example projects.
+Download and install it as per instructions in https://maven.apache.org/[maven.apache.org].
+A Maven WAR project has the following basic structure; not all mentioned resources are required, only the `pom.xml` is required and the rest is optional:
+
+[source]
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:42]: Can .. wasn't sure how, will look.

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 9]
```diff
@@ -1,32 +1,140 @@
...
+The https://jakarta.ee/compatibility/[Jakarta EE Compatible Products] page lists several Jakarta runtimes having a servlet container.
+At least those compatible with "Web Profile" and "Full Platform" have a servlet container.
+Examples are Eclipse GlassFish, IBM Open Liberty and Red Hat WildFly.
+They are available in both "Web Profile" and "Full Profile" flavors.
+
+There are also partial Jakarta runtimes having a servlet container.
+Partial as in, they do not fully implement "Web Profile".
+One example is Apache Tomcat.
+It does implement Jakarta Servlet, Jakarta Pages, Jakarta Expression Language, Jakarta WebSocket, and Jakarta Security.
+But it does not implement Jakarta REST, Jakarta CDI, Jakarta Faces nor Jakarta Tags.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:103]: Really?

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 10]
```diff
@@ -1,32 +1,140 @@
...
-* `hello2`, a servlet-based web application in which the components are implemented by two servlet classes
+----
+http(s)://host:port/context-path
+----
 
-The applications are used to illustrate tasks involved in packaging, deploying, configuring, and running an application that contains web components.
+By default, the `context-path` is represented by the base file name of the WAR file, without the extension.
+If there is no web component listening on the root of the context path, then you could face a HTTP 404 'Not Found' error page.
+In that case you would need to use a more specific URL, depending on the configuration of the desired web component.
+This will be detailed in the chapter associated with the example project.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:140]: webapp004.adoc "Mapping URLs to Web Components", unsure how exactly to link.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 11]
```diff
@@ -1,46 +1,52 @@
...
-The client sends an HTTP request to the web server.
-A web server that implements Jakarta Servlet and Jakarta Server Pages technology converts the request into an `HTTPServletRequest` object.
-This object is delivered to a web component, which can interact with JavaBeans components or a database to generate dynamic content.
-The web component can then generate an `HTTPServletResponse` or can pass the request to another web component.
-A web component eventually generates a `HTTPServletResponse` object.
-The web server converts this object to an HTTP response and returns it to the client.
+A service-oriented web application generates dynamic data structures in response to HTTP requests.
+The response is usually represented as a JSON object or as a XML document.
+Service-oriented applications are often directly used by presentation-oriented web applications.
+Development of service-oriented web applications is covered in [TBD].
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:18]: Done but in a modified form (JAX-WS link dropped).

File: src/main/asciidoc/webapp/webapp001.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:12]: Done, but in a modified form: it was first Faces then Servlet now the other way round.

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 12]
```diff
@@ -1,32 +1,140 @@
...
+
+There are also partial Jakarta runtimes having a servlet container.
+Partial as in, they do not fully implement "Web Profile".
+One example is Apache Tomcat.
+It does implement Jakarta Servlet, Jakarta Pages, Jakarta Expression Language, Jakarta WebSocket, and Jakarta Security.
+But it does not implement Jakarta REST, Jakarta CDI, Jakarta Faces nor Jakarta Tags.
 
-. Develop the web application deployment descriptor, if necessary.
+Deploying a WAR file is generally done by placing the WAR file in the runtime-specific folder dedicated for (automatic) deployments.
+The exact location depends on the runtime used:
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:106]: I wanted to be runtime agnostic. I just gave a few examples of most popular ones to give a general idea. But I'll add a phrase "refer to the documentation for your servlet container" as well.

[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:106]: > Does the example deploy to anything via maven

No. Not yet. I had in mind to add at least 3 profiles which download the server and then start/deploy it but that's going to be a bit of work.

File: src/main/asciidoc/webapp/webapp004.adoc [hunk 13]
```diff
@@ -1,202 +1,94 @@
-== A Web Module That Uses Jakarta Servlet Technology: The hello2 Example
+== Hello World Web Application Using Jakarta Servlet
 
-The `hello2` application is a web module that uses Jakarta Servlet technology to display a greeting and response.
-You can use a text editor to view the application files, or you can use NetBeans IDE.
+The `hello-servlet` application is a web application that uses Jakarta Servlet take a name as an HTTP request parameter, generate the greeting and return it as an HTTP response.
+As a service-oriented web application, the response is represented as plain text.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp004.adoc:4]: Response is not HTML, but point taken, I'll clarify.

File: src/main/asciidoc/webapp/webapp002.adoc [hunk 14]
```diff
@@ -1,32 +1,140 @@
...
-Developing web component code is covered in the later chapters.
-Steps 2 through 6 are expanded on in the following sections and illustrated with a Hello, World–style, presentation-oriented application.
-This application allows a user to enter a name into an HTML form and then displays a greeting after the name is submitted.
+You can also use an integrated development environment (IDE) to manage runtimes and deployments.
+Examples are Eclipse IDE, IntelliJ IDEA and Apache NetBeans.
+They can also automatically build the projects for you.
 
-The Hello application contains two web components that generate the greeting and the response.
-This chapter discusses the following simple applications:
+==== Running The Example Projects
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:127]: Examples have not been updated yet.

**Comment by BalusC** [comment]
@kito: your comments have been addressed so far possible: https://github.com/virtua-tech/jakartaee-tutorial/pull/24/commits/d40b774ca836138c789bbc0c6dd3aeae6434a5a8

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:42]: Sorry, not really sure how to callout from a folder structure in ASCII art.

File: src/main/asciidoc/webapp/webapp001.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp001.adoc:16]: doh! you're right.

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:42]: yeah, me neither...

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:103]: man, I was not doing a good job reviewing. I don't know what I was thinking.

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:144]: @jhammen  can you help here?

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:42]: Let's leave it as-is for now, then.

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:131]: Just updated the tech/approach. Previous examples were written with mindset of previous decade.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 15]
```diff
@@ -1,46 +1,52 @@
...
-In the Jakarta EE platform, web components provide the dynamic extension capabilities for a web server.
-Web components can be Jakarta servlets, web pages implemented with Jakarta Faces technology, web service endpoints, or Jakarta Server Pages.
-xref:jakarta-web-application-request-handling[xrefstyle=short] illustrates the interaction between a web client and a web application that uses a servlet.
-The client sends an HTTP request to the web server.
-A web server that implements Jakarta Servlet and Jakarta Server Pages technology converts the request into an `HTTPServletRequest` object.
-This object is delivered to a web component, which can interact with JavaBeans components or a database to generate dynamic content.
-The web component can then generate an `HTTPServletResponse` or can pass the request to another web component.
-A web component eventually generates a `HTTPServletResponse` object.
-The web server converts this object to an HTTP response and returns it to the client.
+A service-oriented web application generates dynamic data structures in response to HTTP requests.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:15]: And added `(also called a "website")` to the presentation-oriented one.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 16]
```diff
@@ -1,46 +1,52 @@
...
+Jakarta Servlet intends to reduce the boilerplate code needed to convert the HTTP request into a Java object and to offer a Java object as a HTTP response, and to manage all the lifecycle around them.
+
+Jakarta Faces is a component based MVC framework that can be executed on a servlet container and act as a presentation-oriented web application.
+Jakarta Faces intends to reduce the boilerplate code needed to collect the request parameters, convert and validate them, update bean properties with them, invoke bean methods, and generate HTML output as response.
+Jakarta Faces is designed as such that it can be exuected on a non-servlet container such as a portlet container.
 
-Web components are supported by the services of a runtime platform called a web container.
-A web container provides such services as request dispatching, security, concurrency, and lifecycle management.
-A web container also gives web components access to such APIs as naming, transactions, and email.
+Jakarta REST is a RESTful web service framework that can be executed on a servlet container and act as a service-oriented web application.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:41]: Yes. Unsure how to improve tho. I've for now remapped subsequent occurrences to 'It'

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 17]
```diff
@@ -1,46 +1,52 @@
...
-A web container provides such services as request dispatching, security, concurrency, and lifecycle management.
-A web container also gives web components access to such APIs as naming, transactions, and email.
+Jakarta REST is a RESTful web service framework that can be executed on a servlet container and act as a service-oriented web application. 
+Jakarta REST intends to reduce the boilerplate code needed to convert the request parameters to a bean and further convert it to the desired response such as JSON or XML.
+Jakarta REST is designed as such that it can be exuected on a non-servlet container such as a microservice.
 
-Certain aspects of web application behavior can be configured when the application is installed, or deployed, to the web container.
-The configuration information can be specified using Jakarta EE annotations or can be maintained in a text file in XML format called a web application deployment descriptor (DD).
-A web application DD must conform to the schema described in the Jakarta Servlet specification.
+A servlet container is a kind of a web container that can be part of a Jakarta runtime such as an application server.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:45]: Yes indeed. It was renamed to "servlet container". This sentence is now the definition of "servlet container" itself.

File: src/main/asciidoc/webapp/webapp001.adoc [hunk 18]
```diff
@@ -1,46 +1,52 @@
...
+Certain aspects of web component behavior can be configured when the web application is installed, or deployed, to a servlet container.
+The configuration information can be specified using annotations or can be maintained in a XML file called a deployment descriptor.
+A deployment descriptor file must comply to the schema described in the associated specification.
```
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:48]: I have clarified the sentence.

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (jhammen) [reply src/main/asciidoc/webapp/webapp002.adoc:144]: xref:web:webapp/webapp.adoc#_mapping_urls_to_web_components[Mapping URLs to Web Components]

[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:144]: @BalusC ^

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (kito99) [reply src/main/asciidoc/webapp/webapp002.adoc:131]: @BalusC ok, looks like the examples that go with this chapter are finished, right? can you raise a PR for those?

**Review by kito99 — CHANGES_REQUESTED** [review]

File: src/main/asciidoc/webapp/webapp002.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp002.adoc:131]: The examples are only rewritten in the tutorial. They are not rewritten in jakartaee-examples project. I will start today.

File: src/main/asciidoc/webapp/webapp001.adoc (hunk 17 above)
[the remark this answers is not in the document]
Comment (BalusC) [reply src/main/asciidoc/webapp/webapp001.adoc:45]: Ok point taken. I dropped the "is a kind of a web container that" part.

**Review by kito99 — APPROVED** [review]

**Pull request merged**